torch/nn/_functions/thnn/normalization.py

```diff
@@ -135,4 +135,101 @@ def backward(self, grad_output):
         return grad_input
 
 
+class SyncBatchNorm(Function):
+
+    @staticmethod
+    def forward(self, input, weight, bias, running_mean, running_var, eps, momentum, process_group=None):
+        input = input.contiguous()
+
+        # calcualte mean/invstd for input.
+        mean, invstd = torch.batch_norm_stats(input, eps)
+        if torch.distributed.is_initialized():
```

Comment:
If the user used SyncBatchNorm in a single process environment, shouldn't we still want to support that instead of throwing an error?

Reply:
Hmm that would be good to have indeed. Could you instead check for `is_initialized & world_size > 1` and directly use `batch_norm` if the check fails (instead of using the split kernel).